asllib/ASLTypingReference.mld

```diff
@@ -1073,49 +1172,136 @@ Typing of Blocks
 }
 
   {3 TypingRule.Block}
-
+    
     {4 Rule}
-    [annotate_block env stm] is <description>.
+    [annotate_block env return_type s] is the result of annotating the
+    statement [s] in [env].
+
+    {4 Info}
+    A local identifier declared with var, let or constant is in scope from the
+    point immediately after its declaration until the end of the immediately
+    enclosing block.
 
+    From that follows that we can discard the environment at the end of an
+    enclosing block.
+    
     {4 Example}
-    <minimal example>
 
 {2
 Typing of Catchers
 }
-[annotate_catchers env catchers otherwise_opt s_m]
+  
+  {3 TypingRule.CatcherNone}
+  [annotate_catchers env return_type (name_opt, ty, stmt)] is [(name_opt, ty,
```

Comment:
It feels strange to me that `return_type` is not hidden in the local `env`. I might refacto that soon.